docs/root/intro/arch_overview/observability/access_logging.rst

```diff
@@ -17,29 +17,54 @@ logs<envoy_v3_api_field_config.listener.v3.Listener.access_log>`. The listener a
+Note: Some information such as upstream host will not be available yet.
+
+HTTP Router Filter
+******************
+
+For Router Filter, is is possible to enable an upstream access log when a new HTTP request is received, and after successfully establishing a connection
+with the upstream by using :ref:`flush upstream log on upstream stream <envoy_v3_api_field_extensions.filters.http.router.v3.Router.UpstreamAccessLogOptions.flush_upstream_log_on_upstream_stream>`
+Note: In case that the HTTP request involvs retries, a start of request upstream access log will be recorded for each retry.
+
 .. _arch_overview_access_log_periodic:
 
 Periodic access logs
 --------------------
 
-If access log is enabled, then by default it will be reported to the configured sinks at the end of a TCP
-stream, or HTTP request. It is possible to extend this behavior and report access logs periodically, and
-right after the TCP Proxy has successfully established a connection with the upstream, or when a new HTTP
-request has been received by the HTTP connection manager. Note: When recording a log after a new HTTP
-has been received, some information such as upstream host will not be available yet. Reporting access logs
-right after upstream connection establishment does not depend on periodic reporting, and and the other way around.
-
 TCP Proxy
 *********
 
-* Periodic access logs can be enabled using :ref:`access log flush interval <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.access_log_flush_interval>`
-* Access log after upstream connection can be enabled using :ref:`flush access log on connected <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.flush_access_log_on_connected>`
+For TCP Proxy, it is possible to enable a prediodic access log by using
+:ref:`access log flush interval <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.access_log_flush_interval>`
+Note: The first interval count will start right after a new connection received by the TCP Proxy and before making an upstream request.
 
 HTTP Connection Manager
 ***********************
 
-* Periodic access logs can be enabled using :ref:`access log flush interval <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.access_log_flush_interval>`
-* Access log when a new HTTP request is received can be enabled using :ref:`flush access log on new request <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.flush_access_log_on_new_request>`
+For HTTP Connection Manager, it is possible to enable a prediodic access log by using
+:ref:`access log flush interval <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.access_log_flush_interval>`
+Note: The first interval count will start right after the HTTP request is received by the HTTP Connection Manager filter and before iterating the HTTP filter chain.
```

Comment:
If you have both periodic and "flush on new request" enabled, will we do 2 logs on downstream connection or only one?

Reply:
I wrote above the "Start of session access logs" that: "Reporting access logs right after upstream connection establishment or new incoming HTTP request does not depend on periodic reporting, and the other way around." once, otherwise I'd have to repeat this for both sections. Please let me know if further elaboration is required.

Reply:
I don't think it needs to be repeated in each section, but it wasn't immediately clear to me that enabling both features would cause two log messages. (I think it would be nice if we detected both features were on and only logged once at the start, but I haven't looked at the periodic code closely enough to know how difficult that would be.)

Reply:
I see. I added a related comment that I think would make the behavior clear to me.